### PR TITLE
Queue Page

### PR DESCRIPTION
**Pull request by lowtorola**
Adds a queue page!

### CODE REVIEW

**Comment by lowtorola** [comment]
@acrantel I'm having some troubles with styling/controlling my `AsyncSelectMenu` component... think you could take a look and see if I'm missing anything obvious?
1. Input box inside select is outlined
2. Placeholder doesn't show up
3. Can't figure out how to close the menu when you either click away or select an option

**Comment by lowtorola** [comment]
Also my commit history is scuffed... do i need to rebase?

**Comment by lowtorola** [comment]
TODO: remove `ul`s from `BattlecodeTableBottomElement.tsx` 😆

**Comment by lowtorola** [comment]
Current bugs:
- [ ] Placeholder never shows up
- [ ] Input element is outlined
- [ ] entering empty string (manually clearing) doesn't register as non-input
- [ ] selecting option doesn't close dropdown menu
- [ ]

**Comment by lowtorola** [comment]
Note: this blocks `tournaments-page` and `scrimmaging-page`

**Comment by lowtorola** [comment]
TODO: fix tooltip for failed matches in the queue table! (this is for my own reference)

File: frontend2/src/components/EpisodeLayout.tsx [hunk 1]
```diff
@@ -9,9 +9,11 @@ import { EpisodeContext } from "../contexts/EpisodeContext";
...
   const episodeContext = useContext(EpisodeContext);
   const { episodeId } = useParams();
-  if (episodeId !== undefined && episodeId !== episodeContext.episodeId) {
-    episodeContext.setEpisodeId(episodeId);
-  }
+  useEffect(() => {
+    if (episodeId !== undefined && episodeId !== episodeContext.episodeId) {
+      episodeContext.setEpisodeId(episodeId);
+    }
+  }, [episodeId]);
```
Comment (acrantel) [inline frontend2/src/components/EpisodeLayout.tsx:16]: Note: I put this in a useeffect so react doesn't yell at us

**Review by acrantel — APPROVED** [review]
lgtm! 🥳

File: frontend2/src/components/EpisodeLayout.tsx (hunk 1 above)
Comment (lowtorola) [reply frontend2/src/components/EpisodeLayout.tsx:16]: Wheee luv react 😃

**Pull request merged**